### ISSUE TITLE
Unable to create a model with juju 3.3 and 3.4 controllers.

### ISSUE THREAD

**Issue by Vultaire** [juju/terraform-provider-juju#385]
### Description

On Juju 3.3 (3.3/stable, specifically 3.3.0), using a "hello world" plan deployed against a fresh Juju 3.3 LXD controller, I get errors like this:

```
juju_model.hello_world: Creating...
╷
│ Error: Client Error
│ 
│   with juju_model.hello_world,
│   on hello-world-example.tf line 12, in resource "juju_model" "hello_world":
│   12: resource "juju_model" "hello_world" {
│ 
│ Unable to create model, got error: interface conversion: interface {} is nil, not string
╵
```

When I try this using the 3.2/stable version of Juju, it works without problems.

### Urgency

Casually reporting

### Terraform Juju Provider version

0.10.1

### Terraform version

1.6.6

### Terraform Configuration(s)

```terraform
terraform {
  required_providers {
    juju = {
      source  = "juju/juju"
      version = "~> 0.10.0"
    }
  }
}

provider "juju" {}

resource "juju_model" "hello_world" {
  name = "hello-world"
}

resource "juju_application" "hello_world" {
  name  = "hello-world"
  model = juju_model.hello_world.name

  charm {
    name = "ubuntu"
  }
}
```

### Reproduce / Test

```shell
#!/bin/bash
set -x

# Prerequisites:
# * no Juju controller named "lxd"; it gets created below.
# * "cd" to an empty directory, and populate hello_world.tf using the example provided in this ticket.

# FROM WITHIN THE DIRECTORY CONTAINING hello_world.tf:
sudo snap refresh juju --channel 3.3/stable
juju bootstrap --bootstrap-series jammy localhost lxd
rm -rfv terraform.tfstate .terraform*
terraform init
terraform plan
terraform apply -auto-approve
juju destroy-controller --destroy-all-models --no-prompt lxd
```

### Debug/Panic Output

```shell
# TF_LOG set to DEBUG
2024-01-16T14:28:57.217-0800 [INFO]  backend/local: apply calling Apply
2024-01-16T14:28:57.217-0800 [DEBUG] Building and walking apply graph for NormalMode plan
2024-01-16T14:28:57.218-0800 [DEBUG] Resource state not found for node "juju_model.hello_world", instance juju_model.hello_world
2024-01-16T14:28:57.218-0800 [DEBUG] Resource state not found for node "juju_application.hello_world", instance juju_application.hello_world
2024-01-16T14:28:57.218-0800 [DEBUG] ProviderTransformer: "juju_application.hello_world" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/juju/juju"]
2024-01-16T14:28:57.218-0800 [DEBUG] ProviderTransformer: "juju_model.hello_world (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/juju/juju"]
2024-01-16T14:28:57.218-0800 [DEBUG] ProviderTransformer: "juju_application.hello_world (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/juju/juju"]
2024-01-16T14:28:57.218-0800 [DEBUG] ProviderTransformer: "juju_model.hello_world" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/juju/juju"]
2024-01-16T14:28:57.218-0800 [DEBUG] ReferenceTransformer: "juju_application.hello_world (expand)" references: [juju_model.hello_world juju_model.hello_world juju_model.hello_world (expand)]
2024-01-16T14:28:57.218-0800 [DEBUG] ReferenceTransformer: "juju_model.hello_world" references: []
2024-01-16T14:28:57.218-0800 [DEBUG] ReferenceTransformer: "juju_application.hello_world" references: [juju_model.hello_world juju_model.hello_world juju_model.hello_world (expand)]
2024-01-16T14:28:57.218-0800 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/juju/juju\"]" references: []
2024-01-16T14:28:57.218-0800 [DEBUG] ReferenceTransformer: "juju_model.hello_world (expand)" references: []
2024-01-16T14:28:57.219-0800 [DEBUG] Starting graph walk: walkApply
2024-01-16T14:28:57.219-0800 [DEBUG] created provider logger: level=debug
2024-01-16T14:28:57.219-0800 [INFO]  provider: configuring client automatic mTLS
2024-01-16T14:28:57.225-0800 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/juju/juju/0.10.1/linux_amd64/terraform-provider-juju_v0.10.1 args=[".terraform/providers/registry.terraform.io/juju/juju/0.10.1/linux_amd64/terraform-provider-juju_v0.10.1"]
2024-01-16T14:28:57.225-0800 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/juju/juju/0.10.1/linux_amd64/terraform-provider-juju_v0.10.1 pid=106161
2024-01-16T14:28:57.225-0800 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/juju/juju/0.10.1/linux_amd64/terraform-provider-juju_v0.10.1
2024-01-16T14:28:57.251-0800 [INFO]  provider.terraform-provider-juju_v0.10.1: configuring server automatic mTLS: timestamp=2024-01-16T14:28:57.251-0800
2024-01-16T14:28:57.270-0800 [DEBUG] provider: using plugin: version=6
2024-01-16T14:28:57.270-0800 [DEBUG] provider.terraform-provider-juju_v0.10.1: plugin address: address=/tmp/plugin1080530472 network=unix timestamp=2024-01-16T14:28:57.270-0800
2024-01-16T14:28:57.278-0800 [DEBUG] No provider meta schema returned
2024-01-16T14:28:57.502-0800 [DEBUG] provider.terraform-provider-juju_v0.10.1: Marking Computed attributes with null configuration values as unknown (known after apply) in the plan to prevent potential Terraform errors: tf_provider_addr=registry.terraform.io/juju/juju tf_req_id=e171e007-5bd2-2139-35c7-8f3fe1580327 tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/fwserver/server_planresourcechange.go:195 @module=sdk.framework tf_resource_type=juju_model timestamp=2024-01-16T14:28:57.502-0800
2024-01-16T14:28:57.502-0800 [DEBUG] provider.terraform-provider-juju_v0.10.1: marking computed attribute that is null in the config as unknown: @module=sdk.framework tf_attribute_path="AttributeName(\"credential\")" tf_req_id=e171e007-5bd2-2139-35c7-8f3fe1580327 @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/fwserver/server_planresourcechange.go:399 tf_provider_addr=registry.terraform.io/juju/juju tf_resource_type=juju_model tf_rpc=PlanResourceChange timestamp=2024-01-16T14:28:57.502-0800
2024-01-16T14:28:57.502-0800 [DEBUG] provider.terraform-provider-juju_v0.10.1: marking computed attribute that is null in the config as unknown: tf_resource_type=juju_model @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/fwserver/server_planresourcechange.go:399 @module=sdk.framework tf_provider_addr=registry.terraform.io/juju/juju tf_attribute_path="AttributeName(\"id\")" tf_req_id=e171e007-5bd2-2139-35c7-8f3fe1580327 tf_rpc=PlanResourceChange timestamp=2024-01-16T14:28:57.502-0800
2024-01-16T14:28:57.502-0800 [DEBUG] provider.terraform-provider-juju_v0.10.1: marking computed attribute that is null in the config as unknown: @caller=github.com/hashicorp/terraform-plugin-framework@v1.3.5/internal/fwserver/server_planresourcechange.go:399 tf_provider_addr=registry.terraform.io/juju/juju tf_req_id=e171e007-5bd2-2139-35c7-8f3fe1580327 tf_rpc=PlanResourceChange @module=sdk.framework tf_attribute_path="AttributeName(\"type\")" tf_resource_type=juju_model timestamp=2024-01-16T14:28:57.502-0800
juju_model.hello_world: Creating...
2024-01-16T14:28:57.503-0800 [INFO]  Starting apply for juju_model.hello_world
2024-01-16T14:28:57.503-0800 [DEBUG] juju_model.hello_world: applying the planned Create change
2024-01-16T14:28:57.530-0800 [ERROR] provider.terraform-provider-juju_v0.10.1: Response contains error diagnostic: diagnostic_detail="Unable to create model, got error: interface conversion: interface {} is nil, not string" tf_req_id=fa2c346a-2980-f5a8-50d8-e8280693ce98 @module=sdk.proto diagnostic_summary="Client Error" tf_resource_type=juju_model tf_proto_version=6.3 tf_provider_addr=registry.terraform.io/juju/juju tf_rpc=ApplyResourceChange diagnostic_severity=ERROR @caller=github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/diag/diagnostics.go:58 timestamp=2024-01-16T14:28:57.530-0800
2024-01-16T14:28:57.533-0800 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-01-16T14:28:57.533-0800 [ERROR] vertex "juju_model.hello_world" error: Client Error
╷
│ Error: Client Error
│ 
│   with juju_model.hello_world,
│   on hello-world-example.tf line 12, in resource "juju_model" "hello_world":
│   12: resource "juju_model" "hello_world" {
│ 
│ Unable to create model, got error: interface conversion: interface {} is nil, not string
╵
2024-01-16T14:28:57.536-0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-01-16T14:28:57.538-0800 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/juju/juju/0.10.1/linux_amd64/terraform-provider-juju_v0.10.1 pid=106161
2024-01-16T14:28:57.538-0800 [DEBUG] provider: plugin exited
```

**Comment by hmlanigan** [comment]
This has been broken since at least v0.8.0 and is not a regression.

**Comment by hmlanigan** [comment]
This will be fixed in the juju releases: 3.4.0 and 3.3.2
